dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -563,7 +591,56 @@ public void saveAction(final WorkflowAction action, final List<Permission> perms
 	}
 
 	@WrapInTransaction
-	private void saveAction(WorkflowAction action) throws DotDataException, AlreadyExistException {
+	public void saveAction(final String actionId, final String stepId, final User user) {
+
+		WorkflowAction workflowAction = null;
+		WorkflowStep   workflowStep   = null;
+
+		try {
+
+			Logger.debug(this, "Saving (doing the relationship) the actionId: " + actionId + ", stepId: " + stepId);
+
+			workflowAction = this.findAction(actionId, user);
+			workflowStep   = this.findStep  (stepId);
+
+			if (null == workflowAction) {
+
+				Logger.debug(this, "The action: " + actionId + ", does not exists");
+				throw new DoesNotExistException("Workflow-does-not-exists-action");
+			}
+
+			if (null == workflowStep) {
+
+				Logger.debug(this, "The step: " + stepId + ", does not exists");
+				throw new DoesNotExistException("Workflow-does-not-exists-step");
+			}
+
+			this.workFlowFactory.saveAction(workflowAction, workflowStep);
+		} catch (DoesNotExistException  e) {
+
+			throw e;
+		} catch (AlreadyExistException e) {
+
+			Logger.error(WorkflowAPIImpl.class, e.getMessage(), e);
+			throw new DotWorkflowException("Workflow-action-already-exists", e);
+		} catch (DotSecurityException e) {
+
+			Logger.error(WorkflowAPIImpl.class, e.getMessage(), e);
+			throw new DotWorkflowException("Workflow-permission-issue-save-action", e);
+		} catch (Exception e) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)